### PR TITLE
Implement collections (series) of posts

### PR DESCRIPTION
**Pull request by marekweb**
Adds the concept of collections of posts (also known as series, but "collection" is a more general term).

Collections are defined in a new data file called `collections.yaml`:

https://github.com/sourcegraph/learn/blob/731e6b5df2a7d89383feaf2cd6f3609bcb80e0d1/collections.yaml#L1-L6

For any post which is a member of a collection, it can show a "collection view" at the top of the article:

**Future work (probably beyond this particular pull request):**
- improve the collections view, possibly display it in a side column
- add a markdown front-matter flag to enable/disable the collection view
- consider adding an index page for each collection
- allow collections to have a description, image, or other additional fields

**Screenshot**

Screenshot from 2021-07-09 13-57-57

### CODE REVIEW

**Review by ltagliaferri — COMMENTED** [review]
What do you think about using "how to" as part of the collection title for consistency?

Also, what do you think about leading with the video since it is more of a summary of all the different types. So that `collections.yaml` may look like this:

```
collections:
    - title: How To Use Search Patterns on Sourcegraph
# alternatively    - title: How To Search on Sourcegraph
      members:
          - three-ways-to-search-video
          - literal-search-patterns
          - regular-expression-patterns
```

**Comment by marekweb** [comment]
I implemented the changes that you suggested @ltagliaferri.

@Princesso I'm looking forward to your feedback if you have any thoughts.

File: util/loadCollections.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    const posts = await loadAllPosts()
+    const body = await fs.readFile(collectionsFilePath, 'utf-8')
+    const data = yaml.load(body) as { collections: CollectionDefinition[] }
+    console.log(JSON.stringify(data, null, 2))
+    const collections = data.collections.map(collectionDefinition => {
+        const memberPosts = collectionDefinition.members.map(memberSlug => {
+            const memberPost = posts.find(post => post.slug === memberSlug)
+            if (!memberPost) {
+                throw new Error(
+                    `Post not found: "" from collection "${collectionDefinition.slug}" (${collectionDefinition.title})`
```
Comment (marekweb) [inline util/loadCollections.ts:31]:
```suggestion
                    `Post not found: "${post.slug}" from collection "${collectionDefinition.slug}" (${collectionDefinition.title})`
```

**Review by Princesso — APPROVED** [review]
Great Job Marek, 
Looks good to me.

**Pull request merged**